site/pages/api/get-all-templates.api.ts

```diff
@@ -0,0 +1,190 @@
+            if (!!modeToAdd && !modes.includes(modeToAdd)) {
+                modes.push(modeToAdd);
+            }
+
+            severitys.push(consequence.disruptionSeverity);
+
+            switch (consequence.consequenceType) {
+                case "services":
+                    consequence.services.forEach((service) => {
+                        serviceIds.push(service.id.toString());
+                    });
+
+                    consequence.stops?.map((stop) => {
+                        if (!atcoCodeSet.has(stop.atcoCode)) {
+                            atcoCodeSet.add(stop.atcoCode);
+                            stopsAffectedCount++;
+                        }
+                    });
+                    break;
+
+                case "operatorWide":
+                    isOperatorWideCq = true;
+                    consequence.consequenceOperators.forEach((op) => {
+                        disruptionOperators.push(op.operatorNoc);
+                    });
+                    break;
+
+                case "networkWide":
+                    isNetworkWideCq = true;
+                    break;
+
+                case "stops":
+                    consequence.stops?.map((stop) => {
+                        if (!atcoCodeSet.has(stop.atcoCode)) {
+                            atcoCodeSet.add(stop.atcoCode);
+                            stopsAffectedCount++;
+                        }
+                    });
+                    break;
+            }
+        });
+    }
+
+    const status = getTemplateStatus(disruption);
+
+    return {
+        displayId: disruption.displayId,
+        modes,
+        consequenceLength: disruption.consequences ? disruption.consequences.length : 0,
+        status,
+        severity: getWorstSeverity(severitys),
+        serviceIds,
+        operators: disruptionOperators,
+        id: disruption.disruptionId,
+        summary: reduceStringWithEllipsis(disruption.summary, 95),
+        validityPeriods: mapValidityPeriods(disruption),
+        isOperatorWideCq: isOperatorWideCq,
+        isNetworkWideCq: isNetworkWideCq,
+        isLive: isLive,
+        stopsAffectedCount: stopsAffectedCount,
+    };
+};
+
+const getAllTemplates = async (req: GetDisruptionsApiRequest, res: NextApiResponse) => {
+    const session = getSession(req);
+
+    if (!session) {
+        res.status(403);
+        return;
+    }
+
+    const { orgId } = session;
+
+    let templateData = await getTemplatesDataFromDynamo(orgId);
+
+    if (templateData) {
+        templateData = templateData.filter(
+            (item) =>
+                item.publishStatus === PublishStatus.published ||
+                item.publishStatus === PublishStatus.draft ||
+                item.publishStatus === PublishStatus.pendingApproval ||
+                item.publishStatus === PublishStatus.editPendingApproval ||
+                item.publishStatus === PublishStatus.rejected,
```

Comment:
Are these assertions required if we do not have approvals enabled for templates?

Reply:
done